scheduler/preemption.go

```diff
@@ -435,6 +438,52 @@ OUTER:
 	return filteredBestAllocs
 }
 
+// PreemptForDevice tries to find allocations to preempt to meet devices needed
+// This is called once per task when assigning devices to the task
+func (p *Preemptor) PreemptForDevice(ask *structs.RequestedDevice, devAlloc *deviceAllocator) []*structs.Allocation {
+	// First group allocations by priority
+	allocsByPriority := filterAndGroupPreemptibleAllocs(p.jobPriority, p.currentAllocs)
+	neededCount := ask.Count
+	preemptedCount := 0
+	var preemptedAllocs []*structs.Allocation
+
+	for _, grpAllocs := range allocsByPriority {
+		for _, alloc := range grpAllocs.allocs {
+			for _, tr := range alloc.AllocatedResources.Tasks {
+				if len(tr.Devices) > 0 {
```

Comment:
Generally try to reduce an indent by inverting the if and doing a continue:

```
if len(tr.Devices) == 0 {
  continue
}
```

But in the case you can just delete the if and it will work